transform/timestamp.go

```diff
@@ -0,0 +1,102 @@
+package transform
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/buger/jsonparser"
+)
+
+// Timestamp parses and formats timestamp strings using the golang syntax
+func Timestamp(spec *Config, data []byte) ([]byte, error) {
+	for k, v := range *spec.Spec {
+		assertedV, vErr := v.(map[string]interface{})
+		if !vErr {
+			return nil, SpecError(fmt.Sprintf("Warn: Invalid spec. Unable to get value for key: %s", k))
+		}
+		inputFormat, inputErr := assertedV["inputFormat"].(string)
+		if !inputErr {
+			return nil, SpecError(fmt.Sprintf("Warn: Invalid spec. Unable to get \"inputFormat\" for key: %s", k))
+		}
+		outputFormat, outputErr := assertedV["outputFormat"].(string)
+		if !outputErr {
+			return nil, SpecError(fmt.Sprintf("Warn: Invalid spec. Unable to get \"outputFormat\" for key: %s", k))
+		}
+		// check if an array wildcard is present and if it is, treat it the
+		// same as a key with an array
+		if k[len(k)-2] == '*' {
+			k = k[:len(k)-3]
+		}
+		// grab the data
+		dataForV, err := getJSONRaw(data, k, spec.Require)
+		if err != nil {
+			return nil, err
+		}
+		// if the key is missing bail and keep iterating
+		if bytes.Compare(dataForV, []byte("null")) == 0 {
```

Comment:
I'm not sure what the right behavior is here. This differs a bit from other transforms. Line 33 just before this handles the require flag itself. However in this case, if `"require": false` and we get back `null` there's still nothing to parse and format.